crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -19,13 +19,33 @@ use naga::valid::Capabilities;
 use std::sync::{Mutex, PoisonError};
 use thiserror::Error;
 use tracing::{debug, error};
-#[cfg(feature = "shader_format_spirv")]
-use wgpu::util::make_spirv;
 use wgpu::{
     DownlevelFlags, Features, PipelineCompilationOptions,
     VertexBufferLayout as RawVertexBufferLayout,
 };
 
+/// Source of a shader module.
+///
+/// The source will be parsed and validated.
+///
+/// Any necessary shader translation (e.g. from WGSL to SPIR-V or vice versa)
+/// will be done internally by wgpu.
+///
+/// This type is unique to the Rust API of `wgpu`. In the WebGPU specification,
+/// only WGSL source code strings are accepted.
+///
+/// This is roughly equivalent to wgpu::ShaderSource
+#[expect(clippy::large_enum_variant)]
+#[derive(Clone, Debug)]
+pub enum ShaderCacheSource<'a> {
+    /// SPIR-V module represented as a slice of words.
+    SpirV(&'a [u8]),
+    /// WGSL module as a string slice.
+    Wgsl(String),
+    /// Naga module.
+    Naga(naga::Module),
+}
```

Comment:
Reviewer note: this is introduced to avoid taking on a dependency on wgpu just for the ShaderSource struct, but also is a looooot simpler than that struct.